skfda/clustering.py

```diff
@@ -0,0 +1,1180 @@
+"""Clustering Module."""
+
+import numpy as np
+from .grid import FDataGrid
+from .metrics import pairwise_distance, lp_distance, norm_lp
+import matplotlib.pyplot as plt
+from mpldatacursor import datacursor
+import matplotlib.patches as mpatches
+from matplotlib.ticker import MaxNLocator
+import random
+from abc import ABC, abstractmethod
+
+__author__ = "Amanda Hernando Bernabé"
+__email__ = "amanda.hernando@estudiante.uam.es"
+
+
+class ClusteringData(ABC):
+    """Base class to implement clustering algorithms.
+
+    Class from which both :class:`K-Means <fda.clustering.KMeans>` and
+    :class:`Fuzzy K-Means <fda.clustering.FuzzyKMeans>` classes inherit."""
+
+    def __init__(self, max_iter, random_state, p):
+        """Sets the arguments *max_iter* and *random_state* and *p*.
+
+        Args:
+            max_iter (int): Maximum number of iterations of the clustering
+                algorithm.
+            random_state (int, RandomState instance or None): Defaults to 0.
+                Determines random number generation for centroid initialization.
+                Use an int to make the randomness deterministic. See Glossary.
+                https://scikit-learn.org/stable/glossary.html#term-random-state
+            p (int): Identifies the p-norm used to calculate the distance
+                between functions.
+        """
+        if max_iter < 1:
+            raise ValueError(
+                "The number of iterations must be greater than 0.")
+
+        self._max_iter = max_iter
+        self._random_state = random_state
+        self._p = p
+
+    @property
+    def fdatagrid(self):
+        return self._fdatagrid
+
+    @property
+    def n_clusters(self):
+        return self._n_clusters
+
+    @property
+    def init(self):
```

Comment:
This name is very confusing.

Reply:
It is the name used in sklearn Kmeans class.

Reply:
In this case, I guess it is ok, for compatibility.